user-guide/modules/ROOT/pages/advanced-builds.adoc

```diff
@@ -1,3 +1,247 @@
++
+[source,txt]
+----
+./b2 link=dynamic
+./b2 link=shared
+----
+
+Not all Boost libraries need to be linked to your program. Many are "header-only", meaning you only need to include their headers in your source files and don't need to worry about linking, as there is nothing to link. 
+
+== Build Boost with CMake
+
+The only supported method of building the Boost libraries is with the `b2` tool. However, there are experimental alternatives, including building with CMake. Refer to the https://github.com/boostorg/cmake[Boost CMake support infrastructure] for full details, including information on general and library-specific configuration variables, testing after building with CMake, using Boost (or an individual Boost library) with the `add_subdirectory` command, and using Boost with the `FetchContent` command.
+
+== Parallel Builds
+
+To speed up the build process, you might want to utilize parallel compilation. This is where multiple compilation tasks are performed concurrently to speed up the build process. You can enable this feature by using the `-j` option followed by the number of tasks you want to run in parallel when invoking `b2`.
+
+For example, let's say you have a Jamroot.jam file at the root of your project that describes how to build your project. The project consists of several source files spread across different directories, and you are working on a multi-core machine where parallel builds could significantly speed up the build time. To build your project using Boost.Build, you would normally navigate to your project's root directory and run `b2`. However, to take advantage of parallel builds, you can specify the `-j` option followed by the number of tasks you want to run concurrently. If you have a quad-core processor, you might run up to four tasks in parallel:
+
+[source,txt]
+----
+b2 -j4
+----
+
+Running concurrent tasks can increase the system load and may not always result in a linear speedup. However, it is a great idea to use your multiple cores, if you have them available.
+
+== Custom Build Steps
+
+Your application might require pre-processing steps before compiling. For instance, you might need to generate some pass:[C++] code based on a configuration file. Custom build steps in Boost.Build can be achieved using _rule definitions_ and _actions_. Rules in Boost.Build are a little like functions in a programming language. You can define your own rule to execute custom commands, and then invoke the rule at the appropriate point in your Jamfile.
```

Comment:
I'm not even sure I understand this subsection. We went from something very basic (listing how to set build options when only INVOKING b2 to build boost, where the user doesn't have to know anything about b2 at all) to a very very VERY specific and very very VERY advanced topic for someone who's using b2 for their own application AND needs to create custom rules (few people in the world ever needed that) AND all before presenting any other way more basic b2 concepts (which is correct because this should be in the b2 documentation).

Reply:
No user is building their projects through b2. This does not make sense in the user guide.

Reply:
ok, removed this section